src/rpc/mod.rs

```diff
@@ -110,6 +113,41 @@ where
             expected_payment,
         )
     }
+
+    fn check_gas_limit(
+        &self,
+        parent_hash: &B256,
+        registered_gas_limit: u64,
+        block_gas_limit: u64,
+    ) -> RpcResult<()> {
+        let parent =
+            self.provider()
+                .header(parent_hash)
+                .to_rpc_result()?
+                .ok_or(internal_rpc_err(format!(
+                    "Parent block with hash {} not found",
+                    parent_hash
+                )))?;
+
+        // Prysm has a bug where it registers validators with a desired gas limit
+        // of 0. Some builders treat these as desiring gas limit 30_000_000. As a
+        // workaround, whenever the desired gas limit is 0, we accept both the
+        // limit as calculated with a desired limit of 0, and builders which fall
+        // back to calculating with the default 30_000_000.
+        // TODO: Review if we still need this
+        if registered_gas_limit != 0
```

Comment:
How about we have two early return OK cases and err becomes the default last return with Err?

For context: here we still nest an if which feels more complex to me. Second, the first if is looking for a positive match but written as a negative.

Reply:
Makes sense.
Adjusted accordingly in: https://github.com/ultrasoundmoney/reth-payload-validator/pull/18/commits/76a7f23d5fa6cc7da1fe2eacebda1a68727a1bcb